src/runtime-rs/crates/resource/src/rootfs/nydus_rootfs.rs

```diff
@@ -151,3 +160,75 @@ impl Rootfs for NydusRootfs {
         Ok(())
     }
 }
+
+async fn get_nydus_prefetch_files(
+    nydus_image_key: String,
+    nydus_prefetch_path: String,
+) -> Option<String> {
+    // nydus_image_key is an annotation and pod with it will indicate
+    // that prefetch_files will be needed.
+    if nydus_image_key.is_empty() {
+        debug!(
+            sl!(),
+            "nydus image key annotation empty, no need to get prefetch files."
+        );
+        return None;
+    }
+
+    // nydus_prefetch_path is set in configuration.toml's
+    if nydus_prefetch_path.is_empty() {
+        debug!(
+            sl!(),
+            "nydus prefetch file path not set, no prefetch list file found."
+        );
+        return None;
+    }
+
+    let prefetch_list_path = Path::new(nydus_prefetch_path.as_str())
+        .join(nydus_image_key.as_str())
+        .join(NYDUS_PREFETCH_FILE_LIST);
+
+    // ensure the prefetch_list_path is a regular file.
+    if !prefetch_list_path.is_file() {
+        debug!(
+            sl!(),
+            "nydus prefetch list file {:?} not a regular file", &prefetch_list_path
+        );
+
+        return None;
+    }
+
+    return Some(prefetch_list_path.to_str().unwrap().to_string());
```

Comment:
Avoid using `unwrap()`, using `and_then` of an Option object may be ok.

Reply:
> Wait, why is the option object necessary to start with?

Thx, the Option Object chosen as the result of the annotation sometimes maybe empty. And we need ensure the `prefetch_files_list of None` case be handled correctly.